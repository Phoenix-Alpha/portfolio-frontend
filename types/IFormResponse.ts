export interface IFormResponse {
  isSuccess: boolean
  message?: string
}
