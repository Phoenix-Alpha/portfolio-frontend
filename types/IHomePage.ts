export interface IHomePage {
  heroContent?: string
}
