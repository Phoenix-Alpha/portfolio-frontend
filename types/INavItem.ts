export interface INavItem {
  key: string
  title: string
  url?: string
  isGroup?: boolean
  divider?: boolean
  subItems?: INavItem[]
  external?: string
  target?: string
}
