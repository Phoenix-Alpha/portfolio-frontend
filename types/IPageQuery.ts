import { DocumentNode } from 'graphql'
import { AnyAction } from 'redux'

export interface IPageQuery {
  query: DocumentNode
  variables?: Record<string, unknown>
  context?: Record<string, unknown>
  path?: string
  normalizer?: (pathEntity?: unknown) => unknown
  action?: (pathEntity?: unknown) => AnyAction
  actions?: Array<(pathEntity?: unknown) => AnyAction>
}
