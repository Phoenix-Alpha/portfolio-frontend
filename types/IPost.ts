export interface IPost {
  id: string
  title: string
  description: string
  slug: string
  date?: string
  cover?: string
  tags?: string[]
  readTimeMins?: number
  content?: string
}
