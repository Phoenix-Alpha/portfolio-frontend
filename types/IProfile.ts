export interface IProfile {
  firstName: string
  lastName: string
  email: string
  headline?: string
  bio?: string
  avatar?: string
  country?: string
  state?: string
  city?: string
  address?: string
}
