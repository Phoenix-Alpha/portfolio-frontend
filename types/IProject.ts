export interface IProject {
  id: string
  title: string
  description: string
  slug: string
  link?: string
  cover?: string
  tags?: string[]
  content?: string
}
