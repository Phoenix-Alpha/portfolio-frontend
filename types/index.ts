export * from './pages'
export * from './IRawUploadFile'
export * from './IPageQuery'
export * from './IProfile'
export * from './IStoreState'
export * from './TTheme'
export * from './INavItem'
export * from './IFormResponse'
export * from './IMeta'
export * from './IRawProfile'
export * from './IHomePage'
export * from './IRawHomePage'
export * from './IPost'
export * from './IProject'
export * from './IRawPost'
export * from './IRawProject'
