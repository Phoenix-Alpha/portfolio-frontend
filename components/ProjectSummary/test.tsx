import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { IProject } from '@/types'

import { ProjectSummary } from '.'

describe('<ProjectSummary />', () => {
  it('should renders correct <ProjectSummary />', () => {
    const project: IProject = {
      id: '1',
      title: 'Project title',
      description: 'Project description',
      slug: 'project-title'
    }

    const wrapper = shallow(
      <ProjectSummary project={project} pathPrefix="/project" />
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
