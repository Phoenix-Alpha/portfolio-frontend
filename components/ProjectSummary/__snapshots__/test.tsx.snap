// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectSummary /> should renders correct <ProjectSummary /> 1`] = `
<Article
  className="flex flex-col lg:flex-row lg:justify-between lg:items-start py-4 lg:py-8 text-white"
>
  <div
    className="flex flex-col justify-between pr-10"
  >
    <div>
      <PlainLink
        className="text-white"
        href="/project/project-title"
      >
        <h4
          className="mb-4 font-display !text-lg font-bold"
        >
          Project title
        </h4>
      </PlainLink>
      <h5
        className="font-body lg:text-base font-normal text-gray-2 text-md"
      >
        Project description...
      </h5>
    </div>
    <div
      className="flex flex-row flex-wrap items-center mt-4"
    />
  </div>
</Article>
`;
